src/Illuminate/Support/Str.php

```diff
@@ -341,6 +341,38 @@ public static function convertCase(string $string, int $mode = MB_CASE_FOLD, ?st
         return mb_convert_case($string, $mode, $encoding);
     }
 
+    /**
+     * Compresses a string using gzip compression.
+     *
+     * @param  string  $string  The string to compress.
+     * @param  int  $mode  Compression level (1 to 9, defaults to 5).
+     * @return string|false The compressed string, or false on failure.
+     */
+    public static function compress(string $string, int $mode = 5)
+    {
+        // Ensure compression mode is within valid range
```

Comment:
```php
$mode = min(1, max($mode, 9));
```

Reply:
Although I'd prefer it to throw an `\OutOfBoundsException`

https://www.php.net/manual/en/class.outofboundsexception.php

Reply:
According to you, I adjusted the method.